controllers/deactivation/deactivation_controller.go

```diff
@@ -189,7 +188,6 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 
 	// If the usersignup state hasn't been set to deactivating, then set it now
 	if !states.Deactivating(usersignup) {
-		states.SetDeactivating(usersignup, true)
```

Comment:
No, this change was intentional, I should have added a comment in the PR.
I moved it here because of [this](https://github.com/codeready-toolchain/host-operator/blob/master/controllers/deactivation/deactivation_controller_test.go#L299) test. With the fakeClient status handling [change](https://github.com/kubernetes-sigs/controller-runtime/pull/2259) - setting State before resetting the scheduled deactivation time the state would get overriden because SetScheduledDeactivationStatus only updates the status and states are in spec.
Also the comment [here](https://github.com/codeready-toolchain/host-operator/blob/master/controllers/deactivation/deactivation_controller.go#L194)  mentions setting scheduled deactivation status before setting the state, so I moved setting state to after thinking it a better fit and it solved the failing test too.

Reply:
ok, I understand now. I didn't realize that the first update updates the status only. Thanks for the explanation.